converter/src/main/java/org/dswarm/converter/morph/MorphScriptBuilder.java

```diff
@@ -308,6 +310,102 @@ private void createTransformation(final Element rules, final Mapping mapping) th
 		processTransformationComponentFunction(transformationComponent, mapping, mappingInputsVariablesMap, rules);
 	}
 
+	private void createLookupTable(final Element maps, final Component transformationComponent) throws DMPConverterException {
+
+		if (transformationComponent == null)
+			return;
+
+		final Function transformationFunction = transformationComponent.getFunction();
+
+		if (transformationFunction != null && transformationFunction.getFunctionType() == FunctionType.Transformation) {
+
+			final Transformation transformation = (Transformation) transformationFunction;
+
+			final Set<Component> components = transformation.getComponents();
+
+			if (components == null)
+				return;
+
+			for (final Component component : components) {
+
+				final Map<String, String> componentParameterMapping = component.getParameterMappings();
+
+				if (LOOKUP_FUNCTIONS.contains(component.getFunction().getName()) && componentParameterMapping != null) {
+
+					final Element map = doc.createElement("map");
+					maps.appendChild(map);
+
+					for (final Entry<String, String> parameterMapping : componentParameterMapping.entrySet()) {
+
+						if (parameterMapping.getKey() != null && parameterMapping.getValue() != null) {
+
+							switch (parameterMapping.getKey()) {
+
+								case "in":
+
+									map.setAttribute("name", parameterMapping.getValue());
+									break;
+
+								case "map":
+
+									map.setAttribute("name", parameterMapping.getValue());
+									break;
+
+								case LOOKUP_MAP_DEFINITION:
+
+
+									if (component.getFunction().getName().equals("whitelist") || component.getFunction().getName().equals("blacklist")) {
+
+										List<String> lookupList = new ArrayList();
+
+
+										ObjectMapper mapper = new ObjectMapper();
```

Comment:
we may should utilise a class-wide object mapper, instead of always creating a new one. this may increase readability and reduce the lines of code ;)